### DISCUSSION TITLE
Github doesn't respect Folder Name (Upper/Lower Case) changes

### DISCUSSION THREAD

**Discussion by innicoder** [community/community#19561] [Other Feature Feedback, Questions, & Ideas]
One of the issues I have come across is that git and GitHub aren't tracking folder names, be it lowercase or upper case.

Is there a way to solve this or enable it?

**Comment by queenofcorgis** [comment]
Thanks for posting @innicoder :octocat: 

Would you mind elaborating on what you mean by "tracking" so we can better understand your request?

**Reply by innicoder** [reply]
The comment below explains it :)

**Comment by sugar700** [comment]
When using Git on case insensitive file system (for instance when using Windows) it's necessary to rename a file with an intermediary step to make it clear to Git that the file name did change.

For instance, if you have a file called `abc` and want to change it to `ABC` you can do the following.

```sh
git mv abc tmp
git mv tmp ABC
```

**Reply by innicoder** [reply]
Can we fix this? Like, build this fix into git.

**Reply by suvarnak** [reply]
Configuring git to respect case can help git differentiate and track changes, between capital case and lower case folder names.
```
git config core.ignorecase false
```

**Reply by innicoder** [reply]
This needs to be a default settings to `false`.
Folder names are too vast and usually have a reason when they are capital case or lowercase.

Can we put it up for a vote?

**Reply by DevboiDesigns** [reply]
@innicoder upvote for this change

**Comment by DevboiDesigns** [comment]
Could this cause an issue with **Actions** checking out a folder that is lower case in the code (_as well locally_) but GitHub has set it Capital from an older commit. It seems like my Action cannot find the file because of the naming difference?

**Comment by gaffarmalik** [comment]
Yes, this is definitely an issue.
I've been experiencing this on multiple occasions where folder names need to be changed, e.g. Folder named from `Auth` to `auth`. Github ignores the casing an considers it the same.

Does `git config core.ignorecase false` come with any performance degradation?

**Reply by alexiosiko** [reply]
Exacntly what i needed thank you
